extensions/gadgets/base.py

```diff
@@ -84,7 +84,8 @@ def html_body(self):
         once the necessary attributes are supplied.
         """
         html_templates = utils.get_file_contents(os.path.join(
-            feconf.GADGETS_DIR, self.type, '%s.html' % self.type))
+            utils.get_asset_dir_prefix()[1:], feconf.GADGETS_DIR,
```

Comment:
In order to leave gadgets/etc. from this pull requests, I would again need to introduce the folder that would change the name in every new build (so that gadgets files would break the cache)and then the whole system would be useless (because all files would have newq location for every new build). 

But, I have thought about it a little bit yesterday and I have changed the system a little bit so that in backend the files are loaded from _build_staging_ (here the files are build but not renamed), it also simplified few things things in _build.py_

I will try to commit in few hours, or tommorow.